lib/Indexer/Extension/Command/IndexCleanCommand.php

```diff
@@ -52,130 +57,145 @@ protected function configure(): void
+            DOCS, self::OPT_CLEAN_ALL));
+        $this->addArgument(self::ARG_INDEX_NAME, InputArgument::IS_ARRAY, 'Index names to delete');
     }
 
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
-        $indicies = $this->getIndicies();
-        $this->renderIndexTable($indicies, $output);
-
-        $argument = $input->getArgument(self::INDEX_TO_CLEAN);
-        if ($argument === null && !$input->isInteractive()) {
-            return 0;
-        }
-
-        $answer = $argument;
-        if ($answer === null) {
-            $answer = $this->getInteractiveAnswer(count($indicies), $input, $output);
+        if (!$output instanceof ConsoleOutput) {
+            throw new RuntimeException('Symfony');
         }
 
-        $output->writeln('');
+        $indexNames = $input->getArgument(self::ARG_INDEX_NAME);
 
-        foreach ($this->getIndiciesToDelete($indicies, $answer) as $index) {
-            $output->writeln(sprintf('<info>Removing %s</info>', $index->directoryName()));
-            $this->filesystem->remove($index->absolutePath());
+        // Non interactive commands with no index to delete should do nothing
+        if (count($indexNames) === 0 && !$input->isInteractive()) {
+            return 0;
         }
 
-
-        return 0;
-    }
-
-    /**
-     * @param array<IndexInfo> $indecies
-     *
-     * @return array<IndexInfo>
-     */
-    private function getIndiciesToDelete(array $indecies, string $answer): array
-    {
-        $indeciesToDelete = [];
-        if ($answer === self::CLEAN_ALL) {
-            return $indecies;
+        $section = $output->section();
+        $indicies = $this->getIndicies($section);
+        if (count($indexNames) !== 0) {
+            if ($indexNames[0] === self::OPT_CLEAN_ALL) {
+                foreach ($indicies as $index) {
+                    $this->removeIndex($output, $index);
+                }
+                return 0;
+            }
+            foreach ($indexNames as $indexName) {
+                $this->removeIndex($section, $indicies->get($indexName));
+            }
+            return 0;
         }
+        $index = null;
+        $section->clear();
 
-        $indexCount = count($indecies);
-        foreach (explode(',', $answer) as $indexString) {
-            // Trying to find the index by number
-            if (is_numeric($indexString)) {
-                if ($indexCount >= (int) $indexString) {
-                    $indexToDelete = ((int) $indexString) - 1;
+        while (true) {
+            $this->renderIndexTable($indicies, $section);
+            if ($index) {
+                $section->writeln(sprintf('<info>Removed index "%s"</>', $index->name()));
+            }
+            try {
+                // pass $output instead of $section because the interactive input
+                // is corrupted with he Section.
```

Comment:
s/he/the

Reply:
Why do we do the section thing in the first place? Just to reset it? I wouldn't mind just printing the table again.

Reply:
with the section it should keep the text in the same place in the console instead of scrolling off the screen, so looking for like a normal "interface". i think printing it again could look confusing ...